src/pipelinedefinition.cpp

```diff
@@ -477,6 +477,21 @@ class NodeValidator {
         return StatusCode::OK;
     }
 
+    bool isConnectionShapeCorrect(shape_t tensorInputShape, shape_t tensorOutputShape) {
```

Comment:
Here you copy shape_t which is not efficient. It is also possible to have it const since we do not modify it. Please pass it as const reference.